### ISSUE TITLE
VariantFile.subset_samples generates TypeError with Python3

### ISSUE THREAD

**Issue by lbeltrame** [pysam-developers/pysam#272]
```python

In [19]: wanted
Out[19]: ['LC0044-OM']

In [20]: vcf.subset_samples(wanted)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-20-9b7111505aef> in <module>()
----> 1 vcf.subset_samples(wanted)

pysam/cbcf.pyx in pysam.cbcf.VariantFile.subset_samples (pysam/cbcf.c:53336)()

pysam/cbcf.pyx in pysam.cbcf.VariantHeader._subset_samples (pysam/cbcf.c:25980)()

TypeError: sequence item 0: expected a bytes-like object, str found
```

Latest pysam released version, Python 3.5, using a bgzipped + tabix-indexed VCF.

**Comment by lbeltrame** [comment]
Setting the items to be `bytes` makes pysam not find the actual samples:

```ipython
In [21]: wanted = [b'LC0044-OM']

In [22]: vcf.subset_samples(wanted)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-22-9b7111505aef> in <module>()
----> 1 vcf.subset_samples(wanted)

pysam/cbcf.pyx in pysam.cbcf.VariantFile.subset_samples (pysam/cbcf.c:53336)()

pysam/cbcf.pyx in pysam.cbcf.VariantHeader._subset_samples (pysam/cbcf.c:25962)()

ValueError: missing 1 requested samples
```

**Comment by AndreasHeger** [comment]
Thanks, fixed in 6b275c8d8f63c88bf1f321f57e4e5e6d077163ed

**Comment by AndreasHeger** [comment]
Fixed in a452e0eef118f525fd13849da3db90eec4fc87a4
